### PR TITLE
Rework HWID matching/banning system

### PR DESCRIPTION
**Pull request by tsunyoku**

### CODE REVIEW

File: events/loginEvent.py [hunk 1]
```diff
@@ -130,15 +139,6 @@ def handle(
                 glob.verifiedCache[str(userID)] = 0
                 raise exceptions.loginBannedException()
 
-        # Save HWID in db for multiaccount detection
-        hwAllowed = userUtils.logHardware(userID, clientData, firstLogin)
```
Comment (cmyui) [inline events/loginEvent.py:143]: LOL wtf did it not log before if the user got banned?

Comment (tsunyoku) [reply events/loginEvent.py:143]: indeed it did not :pensive:

**Review by infernalfire72 — DISMISSED** [review]

File: events/loginEvent.py [hunk 2]
```diff
@@ -113,6 +113,15 @@ def handle(
 
         """ No login errors! """
 
+        # Save HWID in db for multiaccount detection
+        hwAllowed = userUtils.logHardware(userID, clientData, firstLogin)
```
Comment (cmyui) [inline events/loginEvent.py:126]:
```
Traceback (most recent call last):
  File "/home/akatsuki/pep.py/events/loginEvent.py", line 117, in handle
    hwAllowed = userUtils.logHardware(userID, clientData, firstLogin)
UnboundLocalError: local variable 'firstLogin' referenced before assignment
```

**Comment by cmyui** [comment]
feel free to reopen if you wish to update it

**Pull request closed without merging**